src/peft/tuners/lora/config.py

```diff
@@ -153,5 +153,7 @@ def __post_init__(self):
             raise ValueError("`layers_pattern` cannot be used when `target_modules` is a str.")
 
         if is_megatron_available():
-            if self.tensor_parallel is not None or self.megatron_config is not None:
-                assert self.tensor_parallel is not None and self.megatron_config is not None
+            if (self.tensor_parallel is not None or self.megatron_config is not None) and (
+                self.tensor_parallel is None or self.megatron_config is None
+            ):
```

Comment:
I see the intent now. Couldn't this be simplified to:

`if (self.tensor_parallel is None) is not (self.megatron_config is None):`